File: app/src/main/java/ch/unstable/ost/lists/connection/ConnectionViewHolder.java
package ch.unstable.ost.lists.connection;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import ch.unstable.ost.R;
import ch.unstable.ost.views.ConnectionLineView;

import static com.google.common.base.Preconditions.checkNotNull;
import static com.google.common.base.Verify.verifyNotNull;

/**
 * View holder for connection items
 */
public class ConnectionViewHolder extends RecyclerView.ViewHolder {
    public final TextView startTime;
    public final TextView endTime;
    public final TextView firstEndDestination;
    public final ConnectionLineView connectionLineView;
    public final TextView firstTransportName;
    public final TextView duration;
    public final TextView platform;

    public ConnectionViewHolder(View itemView) {
        super(checkNotNull(itemView));
        startTime = verifyNotNull(itemView.findViewById(R.id.startTime));
        endTime = verifyNotNull(itemView.findViewById(R.id.endTime));
        firstEndDestination = verifyNotNull(itemView.findViewById(R.id.firstSectionEndDestination));
        connectionLineView = verifyNotNull(itemView.findViewById(R.id.connectionLineView));
        firstTransportName = verifyNotNull(itemView.findViewById(R.id.firstTransportName));
        duration = verifyNotNull(itemView.findViewById(R.id.duration));
        platform = verifyNotNull(itemView.findViewById(R.id.platform));
    }
}

File: app/src/main/java/ch/unstable/ost/views/lists/connection/ConnectionViewHolder.kt
package ch.unstable.ost.views.lists.connection

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import ch.unstable.ost.R
import ch.unstable.ost.views.ConnectionLineView

/**
 * View holder for connection items
 */
open class ConnectionViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val differentStartStation: TextView = itemView.findViewById(R.id.differentDepartureStation)
    val startTime: TextView = itemView.findViewById(R.id.startTime)
    val endTime: TextView = itemView.findViewById(R.id.endTime)
    val direction: TextView = itemView.findViewById(R.id.direction)
    val connectionLineView: ConnectionLineView = itemView.findViewById(R.id.connectionLineView)
    val firstTransportName: TextView = itemView.findViewById(R.id.firstTransportName)
    val duration: TextView = itemView.findViewById(R.id.duration)
    val platform: TextView = itemView.findViewById(R.id.platform)

}
